python/pyspark/sql/tests/test_udf.py

```diff
@@ -193,13 +193,14 @@ def check_nondeterministic_udf_in_aggregate(self):
             df.agg(sum(udf_random_col())).collect()
 
     def test_chained_udf(self):
-        self.spark.catalog.registerFunction("double", lambda x: x + x, IntegerType())
```

Comment:
The temp function `double` registered here can not be dropped by `DROP TEMPORARY FUNCTION`
```
org.apache.spark.sql.AnalysisException: Cannot drop built-in function 'double'.
```

and then the temp function fails new test `test_decimal_round`, the `DOUBLE(1.234)` in `spark.sql("SELECT DOUBLE(1.234) AS v")` cause the failure